src/leap/common/events/signals.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# proto.py
+# Copyright (C) 2015 LEAP
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful",
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+
+"""
+Communication protocol definitions.
+"""
+
+
+from enum import Enum
```

Comment:
> we're having troubles because (1) we have to backport it, 

we have to backport it, or not, depending on which `Cryptogrephy`/`pyOpenSSL` version we can/want to use... the problem will be there or not no matter if we discard this dependency for leap* code.

> (2) in some distros it might happen that some other package is requiring the installation of the old enum, which has a different api. 

in an old distro the problem will be which version of `Cryptogrephy`/`pyOpenSSL` we provide? because if we provide a source with bitmask, and a new `pyOpenSSL` which needs enum34 we will have problems anyway.

> all in all, I think we've spent more time dealing with packaging issues than the time it would have taken to code something simple. After all, we only need a simple class to wrap constants.

I agree, my point is not to use `enum34` or a custom one, but that we may want to stick with `enum34` since we will have problems with it anyway.

> cryptography is only needed by pyOpenSSL >= 0.14 if I remember correctly, so it might be the case that in some distros it isn't there.

Does this mean that discarding `enum`/`enum34` we can just ignore the problem and trust the distros to fix this issue for us?